spyder/app/tour.py

```diff
@@ -1164,6 +1209,26 @@ def set_tour(self, index, frames, spy_window):
 
     def start_tour(self):
         """ """
+        self.spy_window.setUpdatesEnabled(False)
+        if self.spy_window.isFullScreen() or self.spy_window.fullscreen_flag:
+            if sys.platform == 'darwin':
+                self.spy_window.setUpdatesEnabled(True)
+                msg_title = _("Request")
+                msg = _("To run the tour, please press the green button in the top left corner of "
+                        "the screen to take Spyder out of fullscreen mode.")
+                QMessageBox.information(self, msg_title, msg,
+                                        QMessageBox.Ok)
+                return
+            else:
+                if self.spy_window.fullscreen_flag:
+                    self.spy_window.toggle_fullscreen()
+                else:
+                    self.spy_window.setWindowState(
+                        self.spy_window.windowState()
+                        & (~ Qt.WindowFullScreen))

```

Comment:
On second thought, this got so long that I think we should put it into another method, e.g. `self._handle_fullscreen()`, which returns `False` if `start_tour` should re-enable window updates and exit (i.e. on macOS) and `True` otherwise.